src/config.rs

```diff
@@ -8,10 +8,59 @@ use serde_json::Value;
 use std::collections::HashMap;
 use std::mem;
 use std::str::FromStr;
+use std::sync::RwLock;
 use wildmatch::WildMatch;
+use lazy_static::lazy_static;
 
 const SETTINGS_JSON: &str = include_str!("./config/settings.json");
 
+struct GlobalStorage {
+    config_store: Option<ConfigStore>,
+}
+
+impl GlobalStorage {
+    pub fn config_store(&mut self) -> &mut ConfigStore {
+        match self.config_store {
+            Some(ref mut store) => store,
+            None => {
+                let store = ConfigStore::from_storage(Box::new(storage::memory_storage::MemoryStorageAdapter::new()), true).unwrap();
+                self.config_store = Some(store);
+                self.config_store.as_mut().unwrap()
+            }
+        }
+    }
+}
+
+lazy_static! {
+    static ref GLOBAL_STORAGE: RwLock<GlobalStorage> = RwLock::new(GlobalStorage {
+        config_store: None,
+    });
+}
+
+#[macro_export]
+macro_rules! config {
+    (string $key:expr) => {
+        let lock = GLOBAL_STORAGE.config_store().read();
```

Comment:
acquire a read lock, do a read, and drop the lock. This will work with a rwlock:  multiple read locks can be acquired, but only one write lock which blocks other locks.